clients/drcachesim/common/options.cpp

```diff
@@ -776,5 +777,56 @@ droption_t<bool> op_enable_kernel_tracing(
     "Processor Trace.");
 #endif
 
+// Core-oriented analysis.
+droption_t<bool> op_core_sharded(
+    DROPTION_SCOPE_ALL, "core_sharded", false, "Analyze per-core in parallel.",
+    "By default, the input trace is analyzed in parallel across shards equal to "
+    "software threads.  This option instead schedules those threads onto virtual cores "
+    "and analyzes each core in parallel.  Thus, each shard consistens of pieces from "
+    "many software threads.  How the scheduling is performed is controlled by a set "
+    "of options with the prefix \"sched_\" along with -num_cores.");
+
+droption_t<bool> op_core_serial(
+    DROPTION_SCOPE_ALL, "core_serial", false, "Analyze per-core in serial.",
+    "In this mode, scheduling is performed just like for -core_sharded. "
+    "However, the resulting schedule is acted upon by a single analysis thread"
+    "which walks the N cores in lockstep in round robin fashion. "
+    "How the scheduling is performed is controlled by a set "
+    "of options with the prefix \"sched_\" along with -num_cores.");
+
+droption_t<int64_t> op_sched_quantum(DROPTION_SCOPE_ALL, "sched_quantum", 1 * 1000 * 1000,
+                                     "Scheduling quantum",
+                                     "Applies to -core_sharded and -core_serial. "
+                                     "Scheduling quantum: in microseconds if -sched_time "
+                                     "is set; otherwise in instructions.");
+
+droption_t<bool>
+    op_sched_time(DROPTION_SCOPE_ALL, "sched_time", false,
+                  "Whether to use time for the scheduling quantum",
+                  "Applies to -core_sharded and -core_serial. "
+                  "Whether to use wall-clock time for the scheduling quantum");
+
+droption_t<bool> op_sched_order_time(DROPTION_SCOPE_ALL, "sched_order_time", true,
```

Comment:
So -core_serial -sched_order_time would be same as -serial?

Reply:
No, there are multiple differences.  `-core_serial` is going to do a dynamic schedule onto N cores just like `-core_sharded`, but it's going to walk them in lockstep in one worker thread -- but still provide the tool with which core is which (replacing drcachesim's -cpu_scheduling).  See above where we decided to not do strict timestamp ordering: the interleaving will not match -serial.  If you replayed as-traced with `-cpu_schedule_file <file> -core_serial` then that is closer to `-serial` but `stream->get_output_cpuid()` will be set for the tool (again for sthg like drcachesim).

Reply:
> If you replayed as-traced with -cpu_schedule_file <file> -core_serial then that is closer to -serial

That should exactly be `-serial` IIUC (except the get_output_cpuid being set), right?

> -core_serial is going to do a dynamic schedule onto N cores

Oh ok, with the dynamic schedule params specified by the op_sched_* flags I understand.